src/openfermion/utils/_commutator_diagonal_coulomb_operator.py

```diff
@@ -0,0 +1,303 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Faster commutators for two-body operators with diagonal Coulomb terms."""
+
+import warnings
+
+from openfermion import FermionOperator, normal_ordered
+
+
+def commutator_ordered_diagonal_coulomb_with_two_body_operator(
+        operator_a, operator_b, result=None):
+    """Compute the commutator of two-body operators provided that both are
+    normal-ordered and that the first only has diagonal Coulomb interactions.
+
+    Args:
+        operator_a: The first FermionOperator argument of the commutator.
+            All terms must be normal-ordered, and furthermore either hopping
+            operators (i^ j) or diagonal Coulomb operators (i^ i or i^ j^ i j).
+        operator_b: The second FermionOperator argument of the commutator.
+            operator_b can be any arbitrary two-body operator.
+        result (optional): The initial FermionOperator to add to.
```

Comment:
why is this called "result"?

Reply:
`resultant` maybe? We're (possibly) taking in some other FermionOperator and adding the commutator to it.

Reply:
resultant isn't any good either. I think its really weird to have an argument called "result". How about something like "prior_terms".

Reply:
I'll change that!